AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -36,7 +36,12 @@ final public class AWSMobileClient: _AWSMobileClient {
     internal var tokenURIQueryParameters: [String: String]? = nil
     internal var signOutURIQueryParameters: [String: String]? = nil
     internal var scopes: [String]? = nil
-    
+
+    // UserPoolOperationHandler should be initialized after AWSMobileClient init and AWSInfo init is
+    // completed, because it require the AWSInfo to be in a valid state to initialize. To acheive this
+    // currently userpoolOpsHelper is initialized inside `_internalInitialize`.
+    internal var userpoolOpsHelper: UserPoolOperationsHandler!
```

Comment:
can we add a note here that this should be accessed synchronously

Reply:
not sure what you meant here?

Reply:
i think UserpoolOpsHelper needs to be accessed in a synchronized way, like in a `initializationqueue.sync` block - i thought it'd be good idea to note it here

Reply:
no, that sync block is for intializing AWSMC, but UserPoolOpsHelper by itself can be accessed from an async context as well.